src/panels/config/automation/action/types/ha-automation-action-choose.ts

```diff
@@ -209,6 +232,46 @@ export class HaChooseAction extends LitElement implements ActionElement {
     `;
   }
 
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+
+    if (changedProps.has("reOrderMode")) {
+      if (this.reOrderMode) {
+        this._createSortable();
+      } else {
+        this._destroySortable();
+      }
+    }
+
+    const nodes = this.shadowRoot!.querySelectorAll("ha-expansion-panel");
+    if (this._expandedStates.length !== nodes.length) {
+      this._expandedStates = [];
+    }
+    let update = false;
+    for (let i = 0; i < nodes.length; i++) {
+      if (this._expandLast && i === nodes.length - 1) {
+        nodes[i].expanded = true;
+        update = true;
+      }
+      if (this._expandedStates[i] !== nodes[i].expanded) {
+        this._expandedStates[i] = nodes[i].expanded;
+        update = true;
+      }
+    }
+    if (update) {
+      this._expandedStates = this._expandedStates.concat();
+    }
+    this._expandLast = false;
```

Comment:
Why do we need all this expanded logic? Doesn't our repeat handle that?

Reply:
It's been a little while so I don't remember exactly what issues I was resolving by doing this, I do remember fighting through a lot of visual glitches with expansion panel to get it working right. I do see one problem if I remove it though. 

If I comment out line 262 (in effect not tracking changes to expanded items), the item titles are not updating correctly. 

Note how for the titles we display the description when the item is collapsed, and we hide the description when the item is expanded (so as not to repeat the same detail twice). 

Now if I remove the tracking, note when I switch option2 for option1, the top item is expanded and the title is shown, and the bottom item is collapsed and the title is not shown (reverse of how it is supposed to be). This state persists for ~6 seconds until the next random render update, in which cases the labels flip to being rendered correctly. 

So I think maybe we need one more render pass after changing the expanded items, since the `isExpanded` function seems to be getting some stale information in the render loop where the items are switched?

![choose-reorder](https://github.com/home-assistant/frontend/assets/32912880/026fa296-dd2b-417b-839b-69e726c58685)

Reply:
Ok I tried rewriting this to track all the expansion states locally instead of querying it from the DOM. Its simpler in some places, more complex in others since we now need to manage tracking the expanded states across adds/deletes/reorders. But it looks like it still works.